imagetest/junit_test.go

```diff
@@ -0,0 +1,141 @@
+package imagetest
+
+import (
+	"testing"
+)
+
+var (
+	testPass = `
+=== RUN   TestUpdateNSSwitchConfig
+--- PASS: TestUpdateNSSwitchConfig (0.00s)
+=== RUN   TestUpdateSSHConfig
+--- PASS: TestUpdateSSHConfig (0.00s)
+=== RUN   TestUpdatePAMsshd
+--- PASS: TestUpdatePAMsshd (0.00s)
+=== RUN   TestUpdateGroupConf
+--- PASS: TestUpdateGroupConf (0.00s)
+PASS
+`
+	testFail = `
+=== RUN   TestAlwaysFails
+--- FAIL: TestAlwaysFails (0.00s)
+    main_test.go:47: failed, message: heh
+    main_test.go:47: failed, message: heh2
+    main_test.go:47: failed, message: heh again
+=== RUN   TestUpdateNSSwitchConfig
+--- PASS: TestUpdateNSSwitchConfig (0.00s)
+=== RUN   TestUpdateSSHConfig
+--- PASS: TestUpdateSSHConfig (0.00s)
+=== RUN   TestUpdatePAMsshd
+--- PASS: TestUpdatePAMsshd (0.00s)
+=== RUN   TestUpdateGroupConf
+--- PASS: TestUpdateGroupConf (0.00s)
+FAIL
+`
+)
+
+func TestConvertToTestSuite(t *testing.T) {
+	tests := []struct {
+		results []string
+		ts      *testSuite
+	}{
+		{
+			[]string{testPass},
+			&testSuite{Tests: 4},
+		},
+		{
+			[]string{testFail},
+			&testSuite{Tests: 5, Failures: 1},
+		},
+		{
+			[]string{testPass, testPass},
+			&testSuite{Tests: 8},
+		},
+		{
+			[]string{testPass, testFail},
+			&testSuite{Tests: 9, Failures: 1},
+		},
+	}
+	for idx, tt := range tests {
+		ts := convertToTestSuite(tt.results)
+		switch {
+		case ts.XMLName != tt.ts.XMLName:
+			fallthrough
```

Comment:
Shouldn't this fail a test?

Reply:
i am handling multiple possible failures with fallthrough to the final case, which does the actual handling. it was an alternative to the comma syntax, but if it's not clear i could switch to the comma syntax instead